io-engine/src/subsys/nvmf/subsystem.rs

```diff
@@ -165,6 +196,189 @@ fn make_sn<T: AsRef<[u8]>>(uuid: T) -> CString {
+            info!(
+                "nvmf_event_handler: controller successfully reset {:?}",
+                success
+            );
+        } else {
+            error!("nvmf_event_handler: failed to reset controller ");
+        }
+        done_cb(ctx, success);
+    }
+
+    pub async fn reset_one_child(dev_name: String, ctx: *mut c_void) {
+        let ctrlr = match NVME_CONTROLLERS.lookup_by_name(&dev_name) {
+            Some(ctrlr) => ctrlr,
+            None => {
+                debug!(
+                    "device name  {:?} not a valid NVMe controller",
+                    dev_name
+                );
+                Self::reset_cb(false, ctx);
+                return;
+            }
+        };
+        debug!("nvmf_event_handler: nvme controller {:?} ", ctrlr);
+        let mut ctrlr = ctrlr.lock();
+        if ctrlr.reset(NvmfSubsystem::reset_cb, ctx, false).is_ok() {
+            debug!("nvmf_event_handler: Reset initiated for {:?} ", ctrlr);
+        } else {
+            error!(
+                "nvmf_event_handler: Failed to initiate reset for {:?} ",
+                ctrlr
+            );
+            Self::reset_cb(false, ctx);
+        }
+    }
+
+    pub async fn reset_nexus(nexus_name: String) {
+        let mut nexus = match nexus_lookup_mut(&nexus_name) {
+            Some(v) => v,
+            None => {
+                debug!("nqn {:?} is not having valid nexus", nexus_name);
+                return;
+            }
+        };
+
+        //Pause the Nexus
+        if nexus.as_mut().pause().await.is_err() {
+            error!("Failed to pause Nexus {:?}", nexus);
+            return;
+        }
+        info!("Paused the nexus - {:?} ", nexus_name);
+        for child in nexus.children_iter() {
+            debug!("Found a {:?} for nexus {:?}", child, nexus_name);
+            if !child.is_healthy() {
+                continue;
+            }
+            let dev = match child.get_device() {
+                Ok(dev) => dev,
+                Err(error) => {
+                    debug!(
+                        "Nexus {:?} get_device failed for {:?} with Error {:?}",
+                        nexus_name, child, error
+                    );
+                    continue;
+                }
+            };
+            let dev_name = dev.device_name();
+            if dev.driver_name() != "nvme" {
+                debug!(
+                    "Device {:?} is not NVMe-oF, reset is not needed",
+                    dev_name
+                );
+                continue;
+            }
+            debug!(
+                "nvmf_event_handler: device name {:?} - nexus nexus_name {:?},
+                 driver {:?}, product {:?}",
+                dev_name,
+                nexus_name,
+                dev.driver_name(),
+                dev.product_name(),
+            );
+            let (sender, receiver) = oneshot::channel::<bool>();
+            Reactors::master().send_future(Self::reset_one_child(
```

Comment:
No need to schedule 'Self::reset_one_child' on Master reactor - 'reset_one_child()' is already running on Master reactor. So you can just call 'Self::reset_one_child' directly using '.await'.

Reply:
I tried this way first, but it didn't work. It got stuck at reset_cb->done_cb

Reply:
Looks like I had a different issue when I tried last time, '::reset_one_child().await works. Thanks for the suggestion.